src/video/cocoa/cocoa_v.mm

```diff
@@ -802,28 +529,31 @@ static void ClearWindowBuffer(uint32 *buffer, uint32 pitch, uint32 height)
 	}
 }
 
-bool WindowQuartzSubdriver::WindowResized()
+/** Resize the window. */
+void VideoDriver_Cocoa::AllocateBackingStore()
 {
-	if (this->window == nil || this->cocoaview == nil) return true;
+	if (this->window == nil || this->cocoaview == nil || this->setup) return;
 
 	NSRect newframe = [ this->cocoaview frame ];
 
 	this->window_width = (int)newframe.size.width;
 	this->window_height = (int)newframe.size.height;
+	this->window_pitch = Align(this->window_width, 16 / sizeof(uint32)); // Quartz likes lines that are multiple of 16-byte.
+	this->buffer_depth = BlitterFactory::GetCurrentBlitter()->GetScreenDepth();
 
 	/* Create Core Graphics Context */
 	free(this->window_buffer);
-	this->window_buffer = malloc(this->window_width * this->window_height * sizeof(uint32));
+	this->window_buffer = malloc(this->window_width * this->window_pitch * sizeof(uint32));
```

Comment:
Feels wrong to me, I think it should be `this->window_pitch * this->window_height`

Reply:
Since width is almost always larger than height, this ends up just overallocating the buffer, unless the player actually makes a tall window. But yes it looks like a bug.